include/merlin/core_kernels/lookup.cuh

```diff
@@ -669,115 +696,124 @@ struct LookupValueBufConfig<Sm70> {
   static constexpr uint32_t size_pipeline_v2 = 64 * sizeof(float);
 };
 
-template <typename K, typename V, typename S = uint64_t,
-          typename ArchTag = Sm80>
+template <typename K, typename V, typename S = uint64_t>
 struct SelectPipelineLookupKernelWithIO {
-  using ValueBufConfig = LookupValueBufConfig<ArchTag>;
-
-  static inline uint32_t max_value_size() {
-    return ValueBufConfig::size_pipeline_v1;
+  static uint32_t max_value_size(int compute_capability) {
+    if (compute_capability >= Sm80::kComputeCapability &&
+        compute_capability != Sm86::kComputeCapability) {
+      return LookupValueBufConfig<Sm80>::size_pipeline_v1;
+    } else {
+      return LookupValueBufConfig<Sm70>::size_pipeline_v1;
+    }
   }
 
   static void select_kernel(LookupKernelParams<K, V, S>& params,
-                            cudaStream_t& stream) {
+                            cudaStream_t& stream, int compute_capability) {
     constexpr int BUCKET_SIZE = 128;
-    constexpr uint32_t buf_size_v1 = ValueBufConfig::size_pipeline_v1;
-    constexpr uint32_t buf_size_v2 = ValueBufConfig::size_pipeline_v2;
 
     uint32_t total_value_size = static_cast<uint32_t>(params.dim * sizeof(V));
 
+    uint32_t size_pipeline_v2;
+
+    if (compute_capability >= Sm80::kComputeCapability &&
+        compute_capability != Sm86::kComputeCapability) {
+      size_pipeline_v2 = LookupValueBufConfig<Sm80>::size_pipeline_v2;
+    } else {
+      size_pipeline_v2 = LookupValueBufConfig<Sm70>::size_pipeline_v2;
```

Comment:
That means the Sm86 use `LookupValueBufConfig<Sm70>::size_pipeline_v2`? It would be better to declare size specially for Sm86, if needed.

Reply:
Yes. The shared memory size of Sm86(100K)  and Sm70(96K) are similar. 
So the template specifiaction of Sm70 also works for Sm72, Sm86.

As for better code readability and consistency, I think we should make a tradeoff between consistency and simplicity?